### PR TITLE
Sample decay time from exponential

### PR DESCRIPTION
**Pull request by hassec**
This replaces the old uniform sampling.

I tested this, and it produces similar/better results.
Efficiencies in generation are better, biases observed in my thesis don't vanish completely.

Testrun was on x=0.0049 and y=0.0061 for 50k events. 100 pseudo experiments.
Pull in x is 0.00+-0.10 and in y -0.20+-0.10

### CODE REVIEW

**Comment by dominikmuller** [comment]
So you are talking about the pull in y?
I might be missing something, but are you sure that bias is significant? +-50% can go away.

**Comment by hassec** [comment]
Hi,

so this is the mean pull on y of 100 different fits.
Therefore it should be zero, no?

**Comment by henryiii** [comment]
@hassec, can you verify that this still works? If so, it can be merged.

File: src/PDFs/Tddp4Pdf.cu [hunk 1]
```diff
@@ -801,23 +827,12 @@ TDDP4::GenerateSig(unsigned int numEvents) {
     unsigned int tmpparam = parameters;
     wmax = maxWeight;
 
+    exp_functor exp_functor_instance {tmpparam, tmpoff, gammamin, wmax};
```
Comment (hassec) [inline src/PDFs/Tddp4Pdf.cu]: this is not needed, right?    
You are using an unnamed instance in l.835 anyway.

Comment (henryiii) [inline src/PDFs/Tddp4Pdf.cu]: Oops, yes. :#)

**Comment by henryiii** [comment]
@hassec, does it look fine otherwise?

**Comment by hassec** [comment]
Yes, LGTM 👍

**Pull request merged**